crates/rust-analyzer/src/config.rs

```diff
@@ -3218,6 +3357,29 @@ fn schema(fields: &[SchemaField]) -> serde_json::Value {
     map.into()
 }
 
+/// Translate a field name to a title case string suitable for use in the category names on the
+/// vscode settings page.
+///
+/// First letter of word should be uppercase, if an uppercase letter is encountered, add a space
+/// before it e.g. "fooBar" -> "Foo Bar", "fooBarBaz" -> "Foo Bar Baz", "foo" -> "Foo"
+///
+/// This likely should be in stdx (or just use heck instead), but it doesn't handle any edge cases
+/// and is intentionally simple.
+fn to_title_case(s: &str) -> String {
```

Comment:
I'm inclined to leave this method here rather than extracting it to something more general. If someone wants to make it more useful, performant etc., then that's fine, but for now it's usefullness is this one use case.

I also don't really care too much about the impl here as it's dev time code. It's likely there's edge cases a more general function would have to catch.